executor/join.go

```diff
@@ -132,7 +133,7 @@ func (e *HashJoinExec) Open(ctx context.Context) error {
 	}
 
 	e.prepared = false
-	e.memTracker = memory.NewTracker(e.id, e.ctx.GetSessionVars().MemQuotaHashJoin)
```

Comment:
Variables like `MemQuotaHashJoin` can be removed in this PR?

Reply:
It’s better to delay this incompatible  deletion(although I think there’re few people to use these useless variables) onto the branch of version 4.0 created so that it has no chance to be affected on version 3.X

Reply:
See https://github.com/pingcap/tidb/issues/12346